app/services/markdown_processor/parser.rb

```diff
@@ -119,7 +119,7 @@ def escape_liquid_tags_in_codeblock(content)
       content.gsub(/[[:space:]]*~{3}.*?~{3}|[[:space:]]*`{3}.*?`{3}|`{2}.+?`{2}|`{1}.+?`{1}/m) do |codeblock|
         codeblock.gsub!("{% endraw %}", "{----% endraw %----}")
         codeblock.gsub!("{% raw %}", "{----% raw %----}")
-        if codeblock.match?(/[[:space:]]*`{3}/)
+        if codeblock.match?(/[[:space:]]*(`{3}|~{3})/)
```

Comment:
this change fixes the 'raw' tag never closed issue I observed when using the original block.